src/common/classes/WipeFile.cpp

```diff
@@ -0,0 +1,659 @@
+}
+
+
+bool SecureDeleteCompressed(const char* FileName) {
+	bool			reallyCompressed = false;
+	bool			zappedFile = false;
+	char			lastFileName[MAX_PATH];
+	static char		volumeName[] = "\\\\.\\A:";
+	static char		volumeRoot[] = "A:\\";
+	HANDLE			hFile;
+	static HANDLE	hVolume = INVALID_HANDLE_VALUE;
+	static DWORD	clusterSize;
+	DWORD			sectorsPerCluster, bytesPerSector, freeClusters, totalClusters;
+
+	// Open the volume name
+	volumeName[4] = FileName[0];
+	hVolume = CreateFile(volumeName, GENERIC_READ | GENERIC_WRITE,
+		FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING,
+		0, 0);
+
+	volumeRoot[0] = FileName[0];
+	GetDiskFreeSpace(volumeRoot, &sectorsPerCluster, &bytesPerSector,
+		&freeClusters, &totalClusters);
+
+	clusterSize = bytesPerSector * sectorsPerCluster;
+
+	hFile = CreateFile(FileName, GENERIC_READ,
+						0,NULL, OPEN_EXISTING, 0, NULL);
+
+	// Scan the location of the file
+	if (!ScanFile(hVolume, clusterSize, hFile, reallyCompressed, zappedFile)) {
+		CloseHandle(hFile);
+		return true;
+	}
+
+	CloseHandle(hFile);
+
+	// If the file is really compressed (it is non-resident),
+	// we can delete it now.
+	if (reallyCompressed) {
+		// Rename the file a few times
+		OverwriteFileName(FileName, lastFileName);
+
+		if (!DeleteFile(lastFileName)) {
+			DWORD error = GetLastError();
+			// Rename back to the original name on error
+			if (!MoveFile( lastFileName, FileName ))
+				fatal_exception::raiseFmt("IO error (%d) deleting file: %s.\nFile is left as %s",
+					error,	FileName, lastFileName);
+			else
+				fatal_exception::raiseFmt("IO error (%d) deleting file: %s",
+					error,	FileName);
+			return true;
+		}
+	}
+
+	// Return TRUE if the file had clusters outside the MFT
+	return reallyCompressed;
+}
+
+
+void SecureDelete(const char* FileName)
+{
+	HANDLE	hFile;
+	ULONGLONG bytesToWrite, bytesWritten;
+	ULARGE_INTEGER fileLength;
+	DWORD dwSizeHigh = 0;
+	DWORD dwSizeLow = 0;
+	char   lastFileName[MAX_PATH];
+
+	// First, open the file in overwrite mode
+	hFile = CreateFile(FileName, GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE,
+					   NULL, OPEN_EXISTING, FILE_FLAG_WRITE_THROUGH, NULL);
+	if (hFile == INVALID_HANDLE_VALUE) {
+		fatal_exception::raiseFmt("IO error (%d) opening file: %s",
+			GetLastError(),	FileName);
+	}
+
+	dwSizeLow = GetFileSize(hFile, &dwSizeHigh);
+
+	// If the file has a non-zero length, fill it with 0's first in order
+	// to preserve is cluster allocation.
+	if (dwSizeLow || dwSizeHigh) {
+
+		// Seek to the last byte of the file
+		dwSizeLow--;
+		if (dwSizeLow == (DWORD) -1 && dwSizeHigh)
+			dwSizeHigh--;
+		SetFilePointer(hFile, dwSizeLow, (PLONG) &dwSizeHigh, FILE_BEGIN);
+
+		// Write one zero byte, which causes the file system to fill the entire
+		// file's on-disk contents with 0.
+		if (!SecureOverwrite(hFile, 1)) {
+			fatal_exception::raiseFmt("IO error (%d) overwriting file: %s",
+				GetLastError(),	FileName);
+			CloseHandle(hFile);
+			return;
+		}
```

Comment:
Fix me: the code after throwing an exception is really supposed to work?..
And why would Windows overwrite whole file because of one byte?..